File: tests/algorithms/x/Java/knapsack/knapsack.java
public class Main {

    static int knapsack(int capacity, int[] weights, int[] values, int counter) {
        if (counter == 0 || capacity == 0) {
            return 0;
        }
        if (weights[counter - 1] > capacity) {
            return knapsack(capacity, ((int[])(weights)), ((int[])(values)), counter - 1);
        } else {
            int left_capacity = capacity - weights[counter - 1];
            int new_value_included = values[counter - 1] + knapsack(left_capacity, ((int[])(weights)), ((int[])(values)), counter - 1);
            int without_new_value = knapsack(capacity, ((int[])(weights)), ((int[])(values)), counter - 1);
            if (new_value_included > without_new_value) {
                return new_value_included;
            } else {
                return without_new_value;
            }
        }
    }

    static void main() {
        int[] weights = ((int[])(new int[]{10, 20, 30}));
        int[] values = ((int[])(new int[]{60, 100, 120}));
        int cap = 50;
        int count = values.length;
        int result = knapsack(cap, ((int[])(weights)), ((int[])(values)), count);
        System.out.println(_p(result));
    }
    public static void main(String[] args) {
        {
            long _benchStart = _now();
            long _benchMem = _mem();
            main();
            long _benchDuration = _now() - _benchStart;
            long _benchMemory = _mem() - _benchMem;
            System.out.println("{");
            System.out.println("  \"duration_us\": " + _benchDuration + ",");
            System.out.println("  \"memory_bytes\": " + _benchMemory + ",");
            System.out.println("  \"name\": \"main\"");
            System.out.println("}");
            return;
        }
    }

    static boolean _nowSeeded = false;
    static int _nowSeed;
    static int _now() {
        if (!_nowSeeded) {
            String s = System.getenv("MOCHI_NOW_SEED");
            if (s != null && !s.isEmpty()) {
                try { _nowSeed = Integer.parseInt(s); _nowSeeded = true; } catch (Exception e) {}
            }
        }
        if (_nowSeeded) {
            _nowSeed = (int)((_nowSeed * 1664525L + 1013904223) % 2147483647);
            return _nowSeed;
        }
        return (int)(System.nanoTime() / 1000);
    }

    static long _mem() {
        Runtime rt = Runtime.getRuntime();
        rt.gc();
        return rt.totalMemory() - rt.freeMemory();
    }

    static String _p(Object v) {
        if (v == null) return "<nil>";
        if (v.getClass().isArray()) {
            if (v instanceof int[]) return java.util.Arrays.toString((int[]) v);
            if (v instanceof long[]) return java.util.Arrays.toString((long[]) v);
            if (v instanceof double[]) return java.util.Arrays.toString((double[]) v);
            if (v instanceof boolean[]) return java.util.Arrays.toString((boolean[]) v);
            if (v instanceof byte[]) return java.util.Arrays.toString((byte[]) v);
            if (v instanceof char[]) return java.util.Arrays.toString((char[]) v);
            if (v instanceof short[]) return java.util.Arrays.toString((short[]) v);
            if (v instanceof float[]) return java.util.Arrays.toString((float[]) v);
            return java.util.Arrays.deepToString((Object[]) v);
        }
        return String.valueOf(v);
    }
}

File: tests/algorithms/x/PHP/knapsack/knapsack.php
<?php
ini_set('memory_limit', '-1');
$now_seed = 0;
$now_seeded = false;
$s = getenv('MOCHI_NOW_SEED');
if ($s !== false && $s !== '') {
    $now_seed = intval($s);
    $now_seeded = true;
}
function _now() {
    global $now_seed, $now_seeded;
    if ($now_seeded) {
        $now_seed = ($now_seed * 1664525 + 1013904223) % 2147483647;
        return $now_seed;
    }
    return hrtime(true);
}
function _str($x) {
    if (is_array($x)) {
        $isList = array_keys($x) === range(0, count($x) - 1);
        if ($isList) {
            $parts = [];
            foreach ($x as $v) { $parts[] = _str($v); }
            return '[' . implode(' ', $parts) . ']';
        }
        $parts = [];
        foreach ($x as $k => $v) { $parts[] = _str($k) . ':' . _str($v); }
        return 'map[' . implode(' ', $parts) . ']';
    }
    if (is_bool($x)) return $x ? 'true' : 'false';
    if ($x === null) return 'null';
    return strval($x);
}
$__start_mem = memory_get_usage();
$__start = _now();
  function knapsack($capacity, $weights, $values, $counter) {
  if ($counter == 0 || $capacity == 0) {
  return 0;
}
  if ($weights[$counter - 1] > $capacity) {
  return knapsack($capacity, $weights, $values, $counter - 1);
} else {
  $left_capacity = $capacity - $weights[$counter - 1];
  $new_value_included = $values[$counter - 1] + knapsack($left_capacity, $weights, $values, $counter - 1);
  $without_new_value = knapsack($capacity, $weights, $values, $counter - 1);
  if ($new_value_included > $without_new_value) {
  return $new_value_included;
} else {
  return $without_new_value;
};
}
};
  function main() {
  $weights = [10, 20, 30];
  $values = [60, 100, 120];
  $cap = 50;
  $count = count($values);
  $result = knapsack($cap, $weights, $values, $count);
  echo rtrim(_str($result)), PHP_EOL;
};
  main();
$__end = _now();
$__end_mem = memory_get_peak_usage();
$__duration = max(1, intdiv($__end - $__start, 1000));
$__mem_diff = max(0, $__end_mem - $__start_mem);
$__bench = ["duration_us" => $__duration, "memory_bytes" => $__mem_diff, "name" => "main"];
$__j = json_encode($__bench, 128);
$__j = str_replace("    ", "  ", $__j);
echo $__j, PHP_EOL;
